### PR TITLE
Add truncation factor for sensitivity weights

### PR DESCRIPTION
**Pull request by dccowan**
Currently, sensitivity weights in directives are stabilized through the 'threshold' property. However, it is difficult to adjust this parameter if one would like to truncate the sensitivity weights. Here we introduce a truncation factor as a final step when creating the sensitivity weights.

### CODE REVIEW

File: SimPEG/directives/directives.py [hunk 1]
```diff
@@ -1669,6 +1670,7 @@ def getWr(self):
 
             wr = wr ** 0.5
             wr /= wr.max()
+            wr[wr < self.truncation_factor] = self.truncation_factor
```
Comment (domfournier) [inline SimPEG/directives/directives.py:1673]: Isn't the same thing as `threshold`, just moved outside of the sum? I would suggest just using the `threshold` parameter for less refactoring required.

Comment (domfournier) [reply SimPEG/directives/directives.py:1673]: they are really doing the same thing, just a smoother corner if you add. 

I am cool with just moving out the threshold out of the sum, but not a fan of adding another term...

Comment (dccowan) [reply SimPEG/directives/directives.py:1673]: I see your point, BUT you don't know if the largest value in diag(JtJ) is 1000 or 0.0001. For example, if I set the threshold to be 1e-6 and my largest diag(JtJ) value is 1e-8, my sensitivity weights are essentially just a constant model of 1e-6.

Whether or not we use threshold or truncation factor, it needs to be done AFTER normalizing and not before!!!

Comment (domfournier) [reply SimPEG/directives/directives.py:1673]: Agreed.

**Comment by jcapriot** [comment]
With the most recent change, the `threshold` value is always relative to the maximum value of the JtJdiag/mesh.cell_volume function. This should make the behavior much easier to understand and use in practice.

Closing this for now. but if you find the new behavior to also be in need of truncating it like this as well, we can discuss further.

**Pull request closed without merging**